### PR TITLE
[FrameworkBundle] access the container getting it from the kernel

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29707
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php [hunk 1]
```diff
@@ -94,7 +94,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $targetArg = rtrim($input->getArgument('target'), '/');
 
         if (!$targetArg) {
-            $targetArg = $this->getPublicDirectory($this->getContainer());
+            $targetArg = $this->getPublicDirectory($kernel->getContainer());
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: alternatively, we could also just inject the `kernel.project_dir` parameter in the constructor

Comment (javiereguiluz) [reply src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: I agree! The `getPublicDirectory()` method only needs the value of `kernel.project_dir`.

Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: On the other hand, custom service definitions will then have to be updated to account for the extra conductor argument. We probably rather want to make that move in 4.3 to be able to issue a proper deprecation.

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: let's discuss this in #29840

**Pull request merged**